packages/core/src/components/portal/portal.tsx

```diff
@@ -52,37 +52,25 @@ export class Portal extends React.Component<IPortalProps, {}> {
 
     private targetElement: HTMLElement;
 
-    public render() {
-        return null as JSX.Element;
-    }
-
-    public componentDidMount() {
-        const targetElement = document.createElement("div");
-        targetElement.classList.add(Classes.PORTAL);
-
-        if (this.context.blueprintPortalClassName != null) {
-            targetElement.classList.add(this.context.blueprintPortalClassName);
+    constructor(props: IPortalProps, context: IPortalContext) {
+        super(props, context);
+        this.targetElement = document.createElement("div");
+        this.targetElement.classList.add(Classes.PORTAL);
```

Comment:
why move to constructor instead of didMount?

Reply:
it happens sooner this way. also this is how the API example was set up

Reply:
oh also I need it to be there by the time `render()` is called, which is not true of `componentDidMount`